### PR TITLE
Force update Microsoft.CSharp transitive dependency

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

* Force update Microsoft.CSharp transitive dependency
  * `NJsonSchema` uses `Namotion.Reflection`, which uses `Microsoft.CSharp 4.3.0` which has a security issue
  * The SDK automatically upgrades this, but some security scanners cannot tell because the `assets.json` and `deps.json` doesn't show evidence of this upgrading behavior.
  * Forcing the upgrade, so that `assets.json` and `deps.json` show 4.7.0, which is sufficient for the vulnerability scanning to work correctly.
* CGManifest looks simpler because we are reading the same data that the scanners are using and now that we only ask for what the SDK has, it does not add the packages to the manifest.

## PR Context

This was done is 7.2:  https://github.com/PowerShell/PowerShell/commit/28514a7ac7c73fcc78ad59341748f66d5333e10b
This was done in 7.3: https://github.com/PowerShell/PowerShell/commit/af8d716ac9bc85a8c35ca2af3afbc6bb88279519

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by daxian-dbw — DISMISSED** [review]
LGTM

File: src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj [hunk 1]
```diff
@@ -36,6 +36,10 @@
     <PackageReference Include="System.Threading.AccessControl" Version="8.0.0-preview.2.23128.3" />
     <PackageReference Include="System.Drawing.Common" Version="8.0.0-preview.2.23128.3" />
     <PackageReference Include="NJsonSchema" Version="10.8.0" />
+    <!-- NJsonSchema brings in an old version of CSharp
+      So, we are forcing a newer version.
+    -->
+    <PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj:42]: With .NET SDK 8.0, we are supposed to use the `Microsoft.Csharp` that is shipped with .NET 8:

image

Will this change brings the old 4.7.0 version of the assembly instead?

Comment (TravisEz13) [reply src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj:42]: No

Comment (daxian-dbw) [reply src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj:42]: Then I'm not sure about the purpose of this change. The `Microsoft.CSharp.dll` appearing in the publish folder without this change is 8.0.0.0, which means `NJsonSchema` doesn't bring in an old version of CSharp for PowerShell build.

Comment (daxian-dbw) [reply src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj:42]: Chatted with Travis offline, and updated the PR description.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Pull request merged**